wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/InternalMachineConfig.java

```diff
@@ -35,12 +35,14 @@
   private final List<InstallerImpl> installers;
   // set of servers including ones configured by installers
   private final Map<String, ServerConfig> servers;
+  private final Map<String, String> env;
   private final Map<String, String> attributes;
 
   InternalMachineConfig(MachineConfig originalConfig, InstallerRegistry installerRegistry)
       throws InfrastructureException {
     this.installers = new ArrayList<>();
     this.servers = new HashMap<>(originalConfig.getServers());
+    this.env = new HashMap<>(originalConfig.getEnv());
```

Comment:
What if getEnv returns null?

Reply:
The same as in case when `originalConfig.getServers` or `originalConfig.getAttributes` return `null`.
Now `originalConfig` doesn't return null values because it is `MachineConfigImpl`.
We can use InternalMachineConfig as constructor argument here or rewrite constructor in the same way as [InternalMachineConfig](https://github.com/eclipse/che/blob/85314523ed5c6d88c6dce68d645bfc04701a92ed/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/MachineConfigImpl.java#L94-L99) has. Which way do you prefer?

Reply:
I've reworked in a second way. Ping me if you think that a first way is better